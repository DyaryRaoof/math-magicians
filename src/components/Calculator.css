.calc-wrapper {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(4, 1fr);
  grid-gap: 1px;
  text-align: center;
  margin: 10% 30%;
}

.calc-wrapper button {
  padding: 10px;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border: 0;
}

.calc-wrapper button:hover {
  opacity: 0.7;
}

.calc-wrapper button:first-child {
  grid-column-start: 1;
  grid-column-end: 5;
  background-color: #858694;
}

.calc-wrapper button:nth-child(18) {
  grid-column-start: 1;
  grid-column-end: 3;
}

.orange {
  background-color: #f5913e;
}

.grey {
  background-color: #e0e0e0;
}
