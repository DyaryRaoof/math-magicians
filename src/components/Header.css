.navbar-wrapper {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}

.navbar-wrapper > span:nth-child(1) {
  font-size: 40px;
  font-weight: bold;
}

.navbar-wrapper > span:not('#logo') {
  padding: 2px;
}
