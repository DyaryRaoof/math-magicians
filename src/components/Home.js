const Home = () => (
  <div>
    <h1>Welcome to our page !</h1>
    <p>
      Non nostrud dolore mollit nostrud fugiat in ut culpa cillum minim tempor.
      Occaecat enim duis Lorem amet aliquip.
      Fugiat officia qui sit consectetur deserunt aute Lorem voluptate.
      Reprehenderit proident laboris irure ut pariatur eu enim minim sunt excepteur.
      Esse fugiat magna occaecat cupidatat nostrud velit consectetur ex ad.
      Ullamco ipsum magna velit ut quis nostrud proident consequat proident
      amet deserunt nostrud dolor nisi. In est fugiat consectetur qui id.
    </p>

    <p>
      Non nostrud dolore mollit nostrud fugiat in ut culpa cillum minim tempor.
      Occaecat enim duis Lorem amet aliquip.
      Fugiat officia qui sit consectetur deserunt aute Lorem voluptate.
      Reprehenderit proident laboris irure ut pariatur eu enim minim sunt excepteur.
      Esse fugiat magna occaecat cupidatat nostrud velit consectetur ex ad.
      Ullamco ipsum magna velit ut quis nostrud proident consequat proident
      amet deserunt nostrud dolor nisi. In est fugiat consectetur qui id.
    </p>
  </div>
);

export default Home;
