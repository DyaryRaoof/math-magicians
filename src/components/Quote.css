.quote-wrapper {
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 10%;
  font-size: 25px;
  font-weight: bold;
}

.activated-link {
  color: orange;
}
