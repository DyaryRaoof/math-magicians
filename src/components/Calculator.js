import { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import CalcButton from './CalcButton';
import DisplayButton from './DisplayButton';

const Caculator = () => {
  const [calculator, setCalculator] = useState({ total: null, next: null, operation: '' });

  const performCalculation = (operation) => {
    setCalculator(calculate(calculator, operation));
  };

  const { total, next, operation } = calculator;

  return (
    <div className="calc-container">
      <h1>Lets do some math</h1>
      <div className="calc-wrapper">
        <DisplayButton total={total} next={next} operation={operation} />
        <CalcButton myClass="grey" myClick={performCalculation} myText="AC" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="+/-" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="%" />
        <CalcButton myClass="orange" myClick={performCalculation} myText="รท" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="7" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="8" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="9" />
        <CalcButton myClass="orange" myClick={performCalculation} myText="x" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="4" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="5" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="6" />
        <CalcButton myClass="orange" myClick={performCalculation} myText="-" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="1" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="2" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="3" />
        <CalcButton myClass="orange" myClick={performCalculation} myText="+" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="0" />
        <CalcButton myClass="grey" myClick={performCalculation} myText="." />
        <CalcButton myClass="orange" myClick={performCalculation} myText="=" />
      </div>
    </div>

  );
};

export default Caculator;
