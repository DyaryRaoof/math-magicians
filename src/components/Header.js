import React from 'react';
import { NavLink } from 'react-router-dom';
import './Header.css';

const Header = () => (
  <div className="navbar-wrapper">
    <span id="logo" data-testid="logo">Math Magicians</span>
    <span>
      <NavLink to="/" exact activeClassName="activated-link">Home</NavLink>
      <span> | </span>
      <NavLink to="/calculator" exact activeClassName="activated-link">Calculator</NavLink>
      <span> | </span>
      <NavLink to="/quote" exact activeClassName="activated-link">Quote</NavLink>
    </span>
  </div>
);

export default Header;
