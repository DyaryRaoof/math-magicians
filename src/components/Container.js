import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Calculator from './Calculator';
import Header from './Header';
import Home from './Home';
import Qoute from './Quote';

const Container = () => (
  <div>
    <Router>
      <Header />
      <Switch>
        <Route exact path="/">
          <Home />
        </Route>
        <Route exact path="/calculator">
          <Calculator />
        </Route>
        <Route exact path="/quote">
          <Qoute />
        </Route>
      </Switch>
    </Router>

  </div>
);

export default Container;
