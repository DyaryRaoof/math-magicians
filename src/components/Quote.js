import './Quote.css';

const Qoute = () => (
  <div className="qoute-wrapper">
    <p>
      Mathematics as an expression of the human mind reflects the active will,
      the contemplative reason, and the desire for aesthetic perfection.
      Its basic elements are logic and intuition, analysis and construction,
      generality and individuality. - Richard Courant
    </p>
  </div>
);

export default Qoute;
