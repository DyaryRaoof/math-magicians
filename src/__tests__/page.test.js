import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';
import Header from '../components/Header';
import Container from '../components/Container';

describe('Testing Home page', () => {
  test('test h1', () => {
    render(<Home />);
    return expect(screen.getByRole('heading')).toBeInTheDocument;
  });
});

describe('Testing Quote page', () => {
  test('test div', () => {
    render(<Quote />);
    return expect(screen.getByTestId('division')).toBeInTheDocument;
  });
});

describe('Testing Calculator page', () => {
  test('test h1', () => {
    render(<Calculator />);
    return expect(screen.getByRole('heading')).toBeInTheDocument;
  });
});

describe('Testing Header page', () => {
  render(
    <Router>
      <Header />
    </Router>,
  );
  const span = screen.getByTestId('logo');
  test('test span', () => expect(span).toBeInTheDocument);
  test('test the logo text', () => expect(span.textContent).toBe('Math Magicians'));
});

describe('Testing Quote page', () => {
  test('test div', () => {
    render(<Container />);
    return expect(screen.getByTestId('container')).toBeInTheDocument;
  });
  test('test h1', () => {
    render(<Container />);
    return expect(screen.getByRole('heading').textContent).toBe('Welcome to our page !');
  });
});

describe('Testing Container page with jest snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Container />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
