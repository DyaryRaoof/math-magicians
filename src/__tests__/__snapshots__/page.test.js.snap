// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Container page with jest snapshot renders correctly 1`] = `
<div
  data-testid="container"
>
  <div
    className="navbar-wrapper"
  >
    <span
      data-testid="logo"
      id="logo"
    >
      Math Magicians
    </span>
    <span>
      <a
        aria-current="page"
        className="activated-link"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <span>
         | 
      </span>
      <a
        aria-current={null}
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <span>
         | 
      </span>
      <a
        aria-current={null}
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </span>
  </div>
  <div>
    <h1>
      Welcome to our page !
    </h1>
    <p>
      Non nostrud dolore mollit nostrud fugiat in ut culpa cillum minim tempor. Occaecat enim duis Lorem amet aliquip. Fugiat officia qui sit consectetur deserunt aute Lorem voluptate. Reprehenderit proident laboris irure ut pariatur eu enim minim sunt excepteur. Esse fugiat magna occaecat cupidatat nostrud velit consectetur ex ad. Ullamco ipsum magna velit ut quis nostrud proident consequat proident amet deserunt nostrud dolor nisi. In est fugiat consectetur qui id.
    </p>
    <p>
      Non nostrud dolore mollit nostrud fugiat in ut culpa cillum minim tempor. Occaecat enim duis Lorem amet aliquip. Fugiat officia qui sit consectetur deserunt aute Lorem voluptate. Reprehenderit proident laboris irure ut pariatur eu enim minim sunt excepteur. Esse fugiat magna occaecat cupidatat nostrud velit consectetur ex ad. Ullamco ipsum magna velit ut quis nostrud proident consequat proident amet deserunt nostrud dolor nisi. In est fugiat consectetur qui id.
    </p>
  </div>
</div>
`;
